### PR TITLE
Added the script to pull annotations from GEO.

### PR DESCRIPTION
**Pull request by bswhite**

### CODE REVIEW

File: synAnnotationUtils/get-geo-annotations.R [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      metadata.tbl$url[i] <- url
+    }
+  }
+  
+  d <- dbDisconnect(con)
+}
+
+write.table(file=output.file, metadata.tbl, sep="\t", quote=FALSE, row.names=FALSE, col.names=TRUE)
+
+cat("Successfully completed\n")
```
Comment (kdaily) [inline synAnnotationUtils/get-geo-annotations.R]: use `message` so it goes to stderr, not stdout.

File: synAnnotationUtils/get-geo-annotations.R [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+gse.identifier <- opt$gse
+output.file <- opt$`output-file`
+
+## Log in to synapse
+synapseLogin()
+
+options('download.file.method.GEOquery' = 'libcurl')
+
+## Get the GSE object
+cat(paste0("Retrieving the GSE object for identifier: ", gse.identifier, "\n"))
```
Comment (kdaily) [inline synAnnotationUtils/get-geo-annotations.R]: use `message` so it goes to stderr, not stdout

File: synAnnotationUtils/get-geo-annotations.R [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+suppressPackageStartupMessages(library("GEOquery"))
+suppressPackageStartupMessages(library("SRAdb"))
+suppressPackageStartupMessages(library("plyr"))
+suppressPackageStartupMessages(library("optparse"))
+
+option_list <- list(
+                    make_option(c("--gse"), action="store",
+                                default=NULL,
+                                help="GEO Series accession number (e.g., \"GSE89777\")"),
+                    make_option(c("--output-file"), action="store",
```
Comment (kdaily) [inline synAnnotationUtils/get-geo-annotations.R]: Consider just writing to stdout and not supplying a file name (then use command line redirect to capture). Standard mechanism for command line programs.

**Review by kdaily — CHANGES_REQUESTED** [review]
Needs to move to `bin/` subdirectory as well.

**Comment by kdaily** [comment]
@teslajoy @sgosline @jaeddy 

OK to have an `R` command line script in a Python repo?

**Comment by teslajoy** [comment]
1. If it contributes to data 
https://github.com/Sage-Bionetworks/synapseAnnotations

2. If it contributes to utilizing users to annotate files on synapse 
in this repo (but I can redo the logic in python to keep it a python package)

3. If it is analysis (doubt it just glancing at it) then we need to talk about a new repo.

**Comment by kdaily** [comment]
@teslajoy we moved it out of synapseAnnotations already because it doesn't fit there.

This script is a utility to get metadata from GEO (Gene Expression Omnibus) given a GEO accession number. So in some sense it is an annotation utility, but it's not really a Synapse annotation utility.

**Comment by kdaily** [comment]
Also seems like this is what the Bioconductor package [GEOmetadb](https://www.bioconductor.org/packages/release/bioc/html/GEOmetadb.html) does?

**Comment by bswhite** [comment]
@kdaily As you've probably seen, the script is a fairly simple use of the GEOquery interface.  I don't see that GEOmetadb will simplify it.  Or, more to the point, I don't speak SQL well, which is the GEOmetadb interface.  I find GEOquery's interface more intuitive.

**Comment by bswhite** [comment]
I think I've accommodated @kdaily's suggested changes above--I have output to stderr with message, output the results to stdout as opposed to an output file specified on the command line, and moved the script to bin.

**Comment by kdaily** [comment]
Imdeed you did! Going to pass this off to @sgosline for suitability in this repo.

**Review by kdaily — APPROVED** [review]
My changes were addressed, leaving it up to others to determine suitability in this repository.

**Comment by sgosline** [comment]
I think it's suitable. The only question I have is if we can use this same script to pull down the 'methods' description as well, as we discussed with @jaeddy last week. If so, it might be handy to include that.

**Comment by teslajoy** [comment]
On a separate project... I just wrote it this morning. May be of use. 
Meta(gsm)$data_processing
or $extract_protocol_ch1
via 
http://bioconductor.org/packages/release/bioc/vignettes/GEOquery/inst/doc/GEOquery.html

**Comment by bswhite** [comment]
The script pulls down any metadata associated with a gsm object--that includes the data_processing and extract_protocol_ch1 fields.  That happens implicitly here:
                        ## Extract its metadata
                        md <- Meta(gsm.geo)
                        
                        ## Metadata is a list of lists.
                        ## Go through each entry and concatenate the individual lists using a ';' delimiter
                        as.data.frame(lapply(md, function(entry) paste(entry, collapse=";")))

File: bin/get-geo-annotations.R [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  q(status=1)
+}
+
+if ( is.null(gse.identifier) ) {
+  print_help(parser)
+  q(status=1)
+}
+
+## Log in to synapse
+synapseLogin()
```
Comment (teslajoy) [inline bin/get-geo-annotations.R]: remove synapseclient dependency :)

File: bin/get-geo-annotations.R [hunk 5]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env Rscript
+
```
Comment (teslajoy) [inline bin/get-geo-annotations.R]: add this :) 
add this :) 
usePackage <- function(p) 
{
  if (!is.element(p, installed.packages()[,1]))
    install.packages(p, dep = TRUE)
  require(p, character.only = TRUE)
}

Comment (kkdang) [reply bin/get-geo-annotations.R]: @bswhite ?

Comment (bswhite) [reply bin/get-geo-annotations.R]: I've made @teslajoy 's two suggested changes above, @kkdang .

**Pull request merged**